jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -0,0 +1,415 @@
+        if (classLoader == null) {
+            List<String> runtimeClasspathElements;
+            try {
+                runtimeClasspathElements = project.getRuntimeClasspathElements();
+            } catch (DependencyResolutionRequiredException e) {
+                throw new MojoExecutionException("Error: Class path construction problem.", e);
+            }
+
+            if (runtimeClasspathElements != null) {
+                URL[] runtimeUrls = new URL[runtimeClasspathElements.size()];
+                for (int i = 0; i < runtimeClasspathElements.size(); i++) {
+                    String element = runtimeClasspathElements.get(i);
+                    try {
+                        runtimeUrls[i] = new File(element).toURI().toURL();
+                    } catch (MalformedURLException e) {
+                        throw new MojoExecutionException("Error: Class path construction problem.", e);
+                    }
+                }
+                classLoader = new URLClassLoader(runtimeUrls,
+                        Thread.currentThread().getContextClassLoader());
+            }
+        }
+
+        return classLoader;
+    }
+
+    /**
+     * Return the name of the file in which the schema has to be written.
+     * The default path is: src/main/resources
+     * The default file name is: name of the class + ".schema.json"
+     *
+     * @return The full path name of the schema file
+     */
+    private File getSchemaFile() {
+        File filePath;
+        if (schemaFilePath == null || schemaFilePath.isEmpty()) {
+            filePath = new File("src" + File.separator + "main" + File.separator + "resources");
+        } else {
+            filePath = new File(schemaFilePath);
+        }
+
+        String fileName;
+        if (schemaFileName == null || schemaFileName.isEmpty()) {
+            fileName =  className + ".schema.json";
+        } else {
+            fileName = schemaFileName;
+        }
+
+        return new File(filePath, fileName);
+    }
+
+    /**
+     * Create the JSON Schema generator.
+     * Configuring it the specified options and add the required modules
+     *
+     * @param mapper The mapper that is to be used
+     * @return The configured generator
+     * @throws MojoExecutionException Error exception
+     */
+    private SchemaGenerator createGenerator(ObjectMapper mapper) throws MojoExecutionException {
+        // Start with the generator builder
+        SchemaGeneratorConfigBuilder configBuilder =
+                new SchemaGeneratorConfigBuilder(mapper, getSchemaVersion(), OptionPreset.PLAIN_JSON);
+
+        // Add options when required
+        if (options.length == 0) {
+            setDefaultOptions(configBuilder);
+        } else {
+            setOptionsFromConfig(configBuilder, options);
+        }
+
+        // Register the modules
+        if (modules.length == 0) {
+            configureDefaultModules(configBuilder);
```

Comment:
As per other comment, I'd suggest to rather not include any modules by default.

Reply:
The main question is whether we can identify a default setup that would be sufficient for a majority of the users. My take is take configuring all modules does not hurt when you do not use the association annotations in your code. But you are right you need to have all the modules in your dependencies.

In case there is no good default behavior...let's kick it out.

I have not done that yet.